lib/iris/tests/unit/coord_categorisation/test_add_categorised_coord.py

```diff
@@ -99,7 +99,6 @@ def setUp(self):
             'proleptic_gregorian': np.array(list(range(360, 367)) +
                                             list(range(1, 4))),
             'noleap': np.array(list(range(359, 366)) + list(range(1, 4))),
-            'julian': np.array(list(range(360, 367)) + list(range(1, 4))),
```

Comment:
@bjlittle This is because netcdftime julian datetime objects don't give the correct day of year in the latest version; they always give a value of 1.

In fact, some of the other calendar types have the same issue if they are constructed directly:
```
>>> import netcdftime
>>> ut = netcdftime.utime('days since 1980-12-25', calendar='365_day')
>>> date = ut.num2date(0)
>>> date2 = netcdftime.DatetimeNoLeap(1980, 12, 25)
>>> print repr(date)
netcdftime._netcdftime.DatetimeNoLeap(1980, 12, 25, 0, 0, 0, 0, 2, 359)
>>> print date.dayofyr
359
>>> print date.timetuple()
(1980, 12, 25, 0, 0, 0, 2, 359, -1)
>>> print repr(date2)           
netcdftime._netcdftime.DatetimeNoLeap(1980, 12, 25, 0, 0, 0, 0, -1, 1)
>>> print date2.dayofyr
1
>>> print date2.timetuple()
(1980, 12, 25, 0, 0, 0, -1, 1, -1)
```

I think it's probably worth just commenting out the offending line in our tests and having a comment explaining why. Either that, or pin netcdf4.